monai/data/dataset.py

```diff
@@ -387,6 +387,12 @@ def _cachecheck(self, item_transformed):
             except PermissionError as e:
                 if sys.platform != "win32":
                     raise e
+            except RuntimeError as e:
+                if "Invalid magic number; corrupt file" in str(e):
+                    print(f"Corrupt cache file detected: {hashfile}. Deleting and recomputing.")
```

Comment:
Adding a warning here would be better.
```suggestion
                    warnings.warn(f"Corrupt cache file detected: {hashfile}. Deleting and recomputing.")
```